### PR TITLE
terraform-providers: update script with Go module support

### PR DESCRIPTION
**Pull request by timstott**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/terraform-providers/update-provider [hunk 1]
```diff
@@ -53,25 +73,53 @@ org="$(echo "$provider_source_url" | cut -d '/' -f 4)"
...
 }
-EOF
+EOS
+)"
+
+if [ "$(jq -r '. | keys | index("vendorSha256")' <<< "$provider_data")" != "null" ]; then
+  puts "provider is Go module will build to compute vendorSha256"
+
+  if [ "$(jq -r '.vendorSha256' <<< "$provider_data")" = "null" ]; then
+
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terraform-providers/update-provider:107]:
```suggestion
```

File: pkgs/applications/networking/cluster/terraform-providers/update-provider [hunk 2]
```diff
@@ -53,25 +73,53 @@ org="$(echo "$provider_source_url" | cut -d '/' -f 4)"
...
+  else
+    update_provider '{ "vendorSha256": "0000000000000000000000000000000000000000000000000000" }'
+  fi
 
-original_provider_list="$(mktemp)"
-cat providers.json > "$original_provider_list"
+  build_output="$(mktemp)"
+  set +e
+  nix-build -A "terraform-providers.$provider_name" 2>&1 | tee "$build_output"
+  set -e
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/cluster/terraform-providers/update-provider:116]:
```suggestion
  nix-build -A "terraform-providers.$provider_name" 2>&1  || true | tee "$build_output"
```

I am not 100% sure about this one.

**Comment by zimbatm** [comment]
the update script in https://github.com/NixOS/nixpkgs/pull/104667 is a bit more feature-complete, in case you want to take it

**Comment by timstott** [comment]
Closing in favor of the much more smashing (improved style, more features) #104667

**Pull request closed without merging**